qcodes/instrument_drivers/QDev/QDac_channels.py

```diff
@@ -414,60 +420,63 @@ def _get_status(self, readcurrents=False):
         0-based, ie chan1 is out[0]
         """
 
-        # Status call
+        def validate_version(version_line):
+            if version_line.startswith('Software Version: '):
+                self.version = version_line.strip().split(': ')[1]
+            else:
+                self._wait_and_clear()
+                raise ValueError('unrecognized version line: ' + version_line)
 
-        version_line = self.ask('status')
+        def validate_header(header_line):
+            headers = header_line.lower().strip('\r\n').split('\t')
+            expected_headers = ['channel', 'out v', '', 'voltage range',
+                                'current range']
+            if headers != expected_headers:
+                raise ValueError('unrecognized header line: ' + header_line)
 
-        if version_line.startswith('Software Version: '):
-            self.version = version_line.strip().split(': ')[1]
-        else:
-            self._wait_and_clear()
-            raise ValueError('unrecognized version line: ' + version_line)
+        def parse_line(line):
+            irange_trans = {'hi cur': 1, 'lo cur': 0}
+            vrange_trans = {'X 1': 0, 'X 0.1': 1}
+
+            chan_str, v_str, _, vrange_str, _, irange_str = line.split('\t')
+            chan = int(chan_str)
+            v = float(v_str)
+            v_range = vrange_trans[vrange_str.strip()]
+            i_range = irange_trans[irange_str.strip()]
+            return chan, i_range, v_range, v
 
-        header_line = self.read()
-        headers = header_line.lower().strip('\r\n').split('\t')
-        expected_headers = ['channel', 'out v', '', 'voltage range',
-                            'current range']
-        if headers != expected_headers:
-            raise ValueError('unrecognized header line: ' + header_line)
+        validate_version(self.ask('status'))
+        validate_header(self.read())
 
-        chans = [{} for _ in self.chan_range]
         chans_left = set(self.chan_range)
         while chans_left:
             line = self.read().strip()
             if not line:
                 continue
-            chanstr, v, _, vrange, _, irange = line.split('\t')
-            chan = int(chanstr)
+            chan, i_range, v_range, v = parse_line(line)
 
-            irange_trans = {'hi cur': 1, 'lo cur': 0}
+            channel = self.channels[chan - 1]
+            channel.vrange.cache(v_range)
+            self._update_v_validator(channel, v_range)
+            channel.irange.cache(i_range)
+            channel.v.cache(v * 0.1 if v_range else v)
```

Comment:
isn't is possible to call `self._set_vrange(chan, v_range)` here instead of `self._update_v_validator(channel, v_range)` and `channel.v.cache(v * 0.1 if v_range else v)`? i'm especially worried about duplicating the `v * 0.1` part of the logic.

Reply:
You are speaking from my heart.  I really wanted to refactor this and already started doing it. But I had to undo the refactoring. Now I have been thinking about it for a while and defined the right vocabulary, writing a github comment, almost in love-letter format, to you. After that I realized a good way of refactoring and implemented it ;-).

Reply:
great :)